### PR TITLE
fix(ext/fetch): Guard against invalid URL before its used by reqwest

### PR DESCRIPTION
**Pull request by kamilogorek**
When `reqwest` is checking for the URL validity 

https://github.com/seanmonstar/reqwest/blob/cdbf84feb1d86b8288796631f2120de5363b3ba9/src/async_impl/client.rs#L1544

it's already too late, as it will panic, without a way to intercept the error

https://github.com/seanmonstar/reqwest/blob/cdbf84feb1d86b8288796631f2120de5363b3ba9/src/into_url.rs#L67-L71

Fixes https://github.com/denoland/deno/issues/17154

~_Note: I see CI is failing, will fix it soon_~

### CODE REVIEW

File: ext/fetch/lib.rs [hunk 1]
```diff
@@ -217,6 +217,12 @@ where
   let method = Method::from_bytes(&method)?;
   let url = Url::parse(&url)?;
 
+  // Make sure that we have a valid URI early, as reqwest's `RequestBuilder::send`
+  // internally uses `expect_uri`, which panics instead of returning a usable `Result`.
+  if url.as_str().parse::<Uri>().is_err() {
+    return Err(type_error("Invalid URL"));
+  }
+
```
Comment (bartlomieju) [inline ext/fetch/lib.rs:225]: Suggest to move it to line 260

**Comment by sigmaSd** [comment]
Maybe also consider opening an issue on reqwest, the docs https://github.com/seanmonstar/reqwest/blob/cdbf84feb1d86b8288796631f2120de5363b3ba9/src/async_impl/client.rs#L1488 doesn't mention this panic condition

**Comment by kamilogorek** [comment]
@sigmaSd done https://github.com/seanmonstar/reqwest/pull/1707

**Comment by kamilogorek** [comment]
Aaaand now it passes 😌

**Review by bartlomieju — APPROVED** [review]
LGTM, thanks! Let's land this fix now and update it if reqwest changes

**Pull request merged**